crates/typst-library/src/sys.rs

```diff
@@ -0,0 +1,39 @@
+//! System-related things.
+
+use std::collections::HashMap;
+
+use ecow::EcoString;
+use typst::eval::{Dict, Module, Scope, Value, Version};
+
+/// Arguments for the `sys` module that handle implementation-specific behaviour.
+#[derive(Clone, Default)]
+pub struct SysArguments {
+    /// A number of keyed inputs that can be provided by the platform and will appear
+    /// as `sys.inputs`. The main expected usecase is scripting.
+    pub inputs: HashMap<EcoString, Value>,
+}
+
+/// Hook up all sys definitions.
+pub(super) fn define(global: &mut Scope, args: SysArguments) {
+    global.category("system");
+    global.define_module(module(args));
+}
+
+/// A module with system-related things.
+fn module(args: SysArguments) -> Module {
+    let mut scope = Scope::deduplicating();
+    scope.category("sys");
+    scope.define(
+        "version",
+        Version::from_iter([
+            env!("CARGO_PKG_VERSION_MAJOR").parse::<u32>().unwrap(),
+            env!("CARGO_PKG_VERSION_MINOR").parse::<u32>().unwrap(),
+            env!("CARGO_PKG_VERSION_PATCH").parse::<u32>().unwrap(),
+        ]),
+    );
+    scope.define(
+        "inputs",
+        Dict::from_iter(args.inputs.into_iter().map(|(k, v)| (k.into(), v))),
+    );
```

Comment:
This is a small point, but I have observed that you can get smaller and nicer output from rustfmt by doing something like

```rust
let inputs = Dict::from_iter(...);
scope.define("inputs", inputs);
```

Reply:
I think it's fine this way, since `scope.define("inputs", some_expression)` is already kinda saying something like `inputs = some_expression`, so it's readable.

Reply:
It's not for clarity, just for formatting.